### PR TITLE
Find and replace script

### PR DESCRIPTION
**Pull request by bchiang7**
## Changes

https://github.com/Upstatement/skela-wp-theme/issues/35

Added a script to automate the case-sensitive search and replace required when setting up a new Skela project!

## How To Test

- Run `./bin/rename-theme` and enter your hypothetical new theme name
- Do a global search of `skela` in your repo and make sure the only instances are ones we intended to keep (i.e. the `img` in `front-page.twig`)
- Make sure `./bin/install` succeeds

### CODE REVIEW

File: bin/theme [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+echo "Replacing 'skela' with '$LOWERCASE'"
+replace_lowercase
+
+echo "Replacing 'SKELA' with '$UPPERCASE'"
+replace_uppercase
+
+echo "Replacing 'Skela' with '$TITLECASE'"
+replace_titlecase
```
Comment (mgburns) [inline bin/theme]: There's a missing case here -- multi-word theme names!

For example, if I set a theme name of `The 19th News` I get the following in `.env`:

```
WORDPRESS_TITLE="The19thNews Site"
UPS_DOCK_NAME=The19thNews
```

When I actually want:

```
WORDPRESS_TITLE="The 19th News"
UPS_DOCK_NAME=The 19th News
```

I think the sed search strings may be a bit too broad.

File: README.md [hunk 2]
```diff
@@ -111,7 +101,15 @@ _**NOTE:** If opting out of one or both of these plugins, **remove** the desired
...
-6. Run the install script
+6. In `package.json` and `composer.json`, update repository and author information
+
+7. Run the theme command and follow the prompt, which will do a find an replace of `skela`, `SKELA`, and `Skela` with your desired theme name
+
+   ```shell
+   ./bin/theme
+   ```
+
+8. Run the install command
```
Comment (mgburns) [inline README.md]: `./bin/install` is failing for me on the `Activate WordPress plugins ...` step:

```
Fatal error: Uncaught Error: Class 'The19thNews\Managers\WordPressManager' not found in /var/www/html/wp-content/themes/the-19th-news/functions.php:37
```

**Comment by bchiang7** [comment]
@mgburns I think I covered all cases this time! I tried a fresh find & replace then `./bin/install` in a different directory, but it looks like Docker uses your local directory name to name the containers though, is there a way to name these manually?

image
 
Also semi-related question -- when I run `docker-compose down -v --remove-orphans`, is that supposed to remove the ups dock network as well? I get `Network ups-dock is external, skipping`, and the network is still there in the ups dock dashboard

image

**Comment by mgburns** [comment]
@bchiang7 Everything is working great for me now!

> @mgburns I think I covered all cases this time! I tried a fresh find & replace then `./bin/install` in a different directory, but it looks like Docker uses your local directory name to name the containers (in this case I named my directory `skela-test`). Is there a way to name these manually? Not a big deal though

I believe we could set the `container_name` for all containers in `docker-compose.yml` to fix this:
https://docs.docker.com/compose/compose-file/#container_name

> ~Also semi-related question -- when I run `docker-compose down -v --remove-orphans`, is that supposed to remove the ups dock network as well? I get `Network ups-dock is external, skipping`, and the network is still there in the ups dock dashboard~

Just noting that `Network ups-dock is external, skipping` is expected. That network is created upon installing [Upstatement/ups-dock](https://github.com/Upstatement/ups-dock) and should not be removed when deleting projects.

File: README.md [hunk 3]
```diff
@@ -111,7 +101,15 @@ _**NOTE:** If opting out of one or both of these plugins, **remove** the desired
 
 5. Copy the `.env.sample` into a new `.env` file, and update any necessary `environment` variables in `docker-compose.yml`
```
Comment (mgburns) [inline README.md]: Does the addition of `./bin/theme` replace the need to update `environment` variables in `docker-compose.yml`?

File: README.md [hunk 4]
```diff
@@ -111,7 +101,15 @@ _**NOTE:** If opting out of one or both of these plugins, **remove** the desired
...
 
-6. Run the install script
+6. In `package.json` and `composer.json`, update repository and author information
+
+7. Run the theme command and follow the prompt, which will set up the project with your desired theme name
+
+   ```shell
+   ./bin/theme
```
Comment (mgburns) [inline README.md]: I know it might break our streak of one-word scripts but ... should we be more description with the script name (e.g. `rename-theme`)?

Comment (bchiang7) [reply README.md]: I was thinking the same thing 😅 I'll rename it

File: README.md (hunk 3 above)
Comment (bchiang7) [reply README.md]: yes, good catch!

**Comment by bchiang7** [comment]
@mgburns I think converting this into a template repo is the way to go!

**Pull request merged**